californium-core/src/main/java/org/eclipse/californium/core/coap/OptionSet.java

```diff
@@ -1270,7 +1283,44 @@ public OptionSet removeObserve() {
 	public static boolean isValidObserveOption(final int value) {
 		return value >= 0 && value <= MAX_OBSERVE_NO;
 	}
-
+	
+	/**
+	 * Sets the Code value.
+	 * Returns the current OptionSet object for a fluent API.
+	 * @param value the code
+	 * @return this OptionSet
+	 */
+	public OptionSet setCode(final int value){
+		this.code = value;
+		return this;
+	}
+	
+	/**
+	 * Returns the uint value of the Code option.
+	 * @return the Code or null if not present.
+	 */
+	public Integer getCode(){
+		return this.code;
+	}
+	
+	/**
+	 * Checks if the Code option is present.
+	 * @return true if Code is present, otherwise false.
+	 */
+	public boolean hasCode(){
+		return this.code != null;
+	}
+	
+	/**
+	 * Removes the Code option.
+	 * Returns the current OptionSet object for a fluent API.
+	 * @return this OptionSet.
+	 */
+	public OptionSet removeCode(){
+		this.code = null;
+		return this;
+	}
+	
```

Comment:
I miss a extension of  `public OptionSet addOption(Option option)`
Is this forgotten or intended not to be provided?

Reply:
This is just forgotten.

Reply:
We could change it to CoAP Code, or CoAP Request Code.

Reply:
May be I misunderstand the intention of "code" in OptionSet. FMPOV the OptionSet is for options, so I assumed the "code" is the new option with the value specified in OSCORE 8.1, the "Object-Security Value", and the pending IANA request in "13.1. CoAP Option Numbers Registry". But now it seems to be even no option and may be just put here for convenience.

So:
- how Is the new option (object-security) handled, if not included in OptionSet?
- why is a none option added to the OptionSet?

Reply:
Exactly, just for convenience. Suggestions: add object-sec as a new option and handle CoaP Code as an Other-option.

None option?

Reply:
I prefer to have OptionSet for options and not for a "easy OptionJuggle". May be you consider to change the processing in a way, where options kept separated from other artefacts. A approach, which creates a different Message/Request/Response instance could help to sort the functions in an other way.

Reply:
I like your suggestion and I will make the changes.